3scale/porta
# CMS API:  Drop support for XML requests

**What this PR does / why we need it**:

This PR removes support for `XML` format requests from all CMS endpoints. It also makes `JSON` the standard request format for the CMS API. 

**Which issue(s) this PR fixes** 

[THREESCALE-9433](https://issues.redhat.com/browse/THREESCALE-9433)

**Verification steps** 

The following endpoints should only respond to `JSON` requests. Any other format should return a `406` status.

| Verb | Endpoint | Controller#Action |
| ------- | ------- | ------- |
|GET|/admin/api/cms/sections/:section_id/files|admin/api/cms/files#index|
|GET|/admin/api/cms/sections|admin/api/cms/sections#index|
|POST|/admin/api/cms/sections|admin/api/cms/sections#create|
|GET|/admin/api/cms/sections/new|admin/api/cms/sections#new|
|GET|/admin/api/cms/sections/:id/edit|admin/api/cms/sections#edit|
|GET|/admin/api/cms/sections/:id|admin/api/cms/sections#show|
|PATCH|/admin/api/cms/sections/:id|admin/api/cms/sections#update|
|PUT|/admin/api/cms/sections/:id|admin/api/cms/sections#update|
|DELETE|/admin/api/cms/sections/:id|admin/api/cms/sections#destroy|
|GET|/admin/api/cms/files|admin/api/cms/files#index|
|POST|/admin/api/cms/files|admin/api/cms/files#create|
|GET|/admin/api/cms/files/new|admin/api/cms/files#new|
|GET|/admin/api/cms/files/:id/edit|admin/api/cms/files#edit|
|GET|/admin/api/cms/files/:id|admin/api/cms/files#show|
|PATCH|/admin/api/cms/files/:id|admin/api/cms/files#update|
|PUT|/admin/api/cms/files/:id|admin/api/cms/files#update|
|DELETE|/admin/api/cms/files/:id|admin/api/cms/files#destroy|
|PUT|/admin/api/cms/templates/:id/publish|admin/api/cms/templates#publish|
|GET|/admin/api/cms/templates|admin/api/cms/templates#index|
|POST|/admin/api/cms/templates|admin/api/cms/templates#create|
|GET|/admin/api/cms/templates/:id|admin/api/cms/templates#show|
|PATCH|/admin/api/cms/templates/:id|admin/api/cms/templates#update|
|PUT|/admin/api/cms/templates/:id|admin/api/cms/templates#update|
|DELETE|/admin/api/cms/templates/:id|admin/api/cms/templates#destroy|

**Special notes for your reviewer**:

![](https://media.giphy.com/media/3eKdC7REvgOt2/giphy.gif)
